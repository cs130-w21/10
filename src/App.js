import React from 'react';
import './App.css';
import Navbar from './components/Navigation';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';
import AuthProvider from './services/AuthContext';
import ProtectedRoute from './helpers/ProtectedRoute';
import Login from './components/Login';
import Test from './components/Test/Test';
import Home from './components/Test/Home';
import EditProfilePopup from './components/Profile/EditProfilePopup';
import Profile from './components/Profile';
import MatchPage from './components/MatchPage/MatchPage';
import ProfileCards from "./components/Swipe";
function App() {
  return (
    <div className="App">
        <Router>
          <AuthProvider>
            <Navbar />
            <Switch>
              <Route exact path ='/login' component={Login} />
              <ProtectedRoute exact path='/test' component={Test} />
              <ProtectedRoute exact path='/' component={Home} />
              <ProtectedRoute path='/profile' exact component={Profile} />
              <ProtectedRoute path='/test-edit-profile' exact component={EditProfilePopup} />
              <ProtectedRoute path="/matches" component={MatchPage} />
              <ProtectedRoute path="/swipe" component={ProfileCards} />
            </Switch>
          </AuthProvider>
        </Router>      
      </div>
  );
}


export default App;
