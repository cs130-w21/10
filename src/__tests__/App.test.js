import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '../App';

test('renders capitalist hinge app', () => {
  render(<App />);
  const linkElement = screen.getAllByText(/Capitalist Hinge/i);
  expect(linkElement[0]).toBeInTheDocument();
  const login = screen.getByText(/Login/i);
  expect(login).toBeInTheDocument();
});
test('snapshot test for profile page', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
})