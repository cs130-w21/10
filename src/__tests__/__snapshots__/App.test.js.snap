// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for profile page 1`] = `
<div
  className="App"
>
  <nav>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          className="MuiContainer-root makeStyles-navbarDisplayFlex-11 MuiContainer-maxWidthLg"
        >
          <a
            className="makeStyles-logo-14"
            href="/"
            onClick={[Function]}
          >
            Capitalist Hinge
          </a>
          <nav
            aria-labelledby="main navigation"
            className="MuiList-root makeStyles-navDisplayFlex-12 MuiList-padding"
          >
            <a
              className="makeStyles-linkText-13"
              href="/matches"
              onClick={[Function]}
            >
              <div
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  className="MuiListItemText-root"
                >
                  <span
                    className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Matches
                  </span>
                </div>
              </div>
            </a>
            <a
              className="makeStyles-linkText-13"
              href="/profile"
              onClick={[Function]}
            >
              <div
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  className="MuiListItemText-root"
                >
                  <span
                    className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Profile
                  </span>
                </div>
              </div>
            </a>
            <a
              className="makeStyles-linkText-13"
              href="/"
              onClick={[Function]}
            >
              <div
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex={0}
              >
                <div
                  className="MuiListItemText-root"
                >
                  <span
                    className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
                  >
                    Sign In
                  </span>
                </div>
              </div>
            </a>
          </nav>
        </div>
      </div>
    </header>
  </nav>
  <div>
    <h1>
      Login
    </h1>
    <div
      id="firebaseui-auth-container"
    />
  </div>
</div>
`;
